### PR TITLE
enh(update): check platform requirements on updates API call

### PR DESCRIPTION
**Pull request by kduret**
## Description

check platform requirements on updates API call

**Fixes** MON-12296

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Check API documentation to make an api call

### CODE REVIEW

**Review by smau-centreon — APPROVED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -16943,3 +16943,15 @@ msgstr "Une erreur s'est produite lors de l'application des actions postérieure
 
 msgid "Updates not found"
 msgstr "Les mises à jour n'ont pas été trouvées"
+
+msgid "PHP version %s required (%s installed)"
+msgstr "La version PHP %s est requise (%s installée)"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "La version %s de PHP est requise (%s installée)"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16943,3 +16943,15 @@ msgstr "Une erreur s'est produite lors de l'application des actions postérieure
...
+
+msgid "PHP extension %s not loaded"
+msgstr "L'extension PHP %s n'est pas chargée"
+
+msgid "Error when getting MariaDb version"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Error when getting MariaDB version"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -16943,3 +16943,15 @@ msgstr "Une erreur s'est produite lors de l'application des actions postérieure
...
+msgstr "Erreur lors de la récupération de la version MariaDb"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Erreur lors de la récupération de la version MariaDB"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -16943,3 +16943,15 @@ msgstr "Une erreur s'est produite lors de l'application des actions postérieure
...
+
+msgid "MariaDb version %s required (%s installed)"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "MariaDB version %s required (%s installed)"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -16943,3 +16943,15 @@ msgstr "Une erreur s'est produite lors de l'application des actions postérieure
...
+msgstr "La version MariaDb %s est requise (%s installée)"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "La version %s de MariaDB est requise (%s installée)"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -16943,3 +16943,15 @@ msgstr "Une erreur s'est produite lors de l'application des actions postérieure
 
 msgid "Updates not found"
 msgstr "Les mises à jour n'ont pas été trouvées"
+
+msgid "PHP version %s required (%s installed)"
+msgstr "La version PHP %s est requise (%s installée)"
+
+msgid "PHP extension %s not loaded"
+msgstr "L'extension PHP %s n'est pas chargée"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "L'extension %s de PHP n'est pas chargée"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by kduret — COMMENTED** [review]
create validator directory instead of repository
create global database validator with provider for each type : MariaDBValidator, ...

**Review by jeremyjaouen — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**